### PR TITLE
qa/fs/mixed-clients: specify distros for tests

### PR DESCRIPTION
**Pull request by dparmar18**
The OS Type and OS Version for mixed-clients tests is left blank on teuthology and they appear to be trying to
run ubuntu jammy which is causing failures. We should specify what distros the tests should run on.

Fixes: https://tracker.ceph.com/issues/57280
Signed-off-by: Dhairya Parmar <dparmar@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by dparmar18** [comment]
Tests pass in teuthology: http://pulpito.front.sepia.ceph.com/dparmar-2022-09-13_11:16:41-fs:mixed-clients-main-distro-default-smithi/

File: qa/suites/fs/mixed-clients/distro/ubuntu/latest.yaml [hunk 1]
```diff
@@ -0,0 +1 @@
+.qa/distros/all/ubuntu_20.04.yaml
```
Comment (vshankar) [inline qa/suites/fs/mixed-clients/distro/ubuntu/latest.yaml:1]: Shaman's do additionally build for jammy, isn't it?

Comment (dparmar18) [reply qa/suites/fs/mixed-clients/distro/ubuntu/latest.yaml:1]: I don't think so, cause if they did we should not be getting this kind of error:

```
teuthology.exceptions.VersionNotFoundError: Failed to fetch package version from https://shaman.ceph.com/api/search/?status=ready&project=kernel&flavor=default&distros=ubuntu%2F22.04%2Fx86_64&ref=testing

Comment (vshankar) [reply qa/suites/fs/mixed-clients/distro/ubuntu/latest.yaml:1]: Well https://shaman.ceph.com/api/search/?status=ready&project=kernel&flavor=default&distros=ubuntu%2F22.04%2Fx86_64&ref=testing does show a valid build.

Comment (dparmar18) [reply qa/suites/fs/mixed-clients/distro/ubuntu/latest.yaml:1]: Yeah but if you look at this log http://qa-proxy.ceph.com/teuthology/yuriw-2022-08-22_21:15:31-fs-wip-yuri7-testing-2022-08-17-0943-quincy-distro-default-smithi/6986427/teuthology.log, `client.0` SHA is `none`:

```
2022-08-22T22:50:57.070 DEBUG:teuthology.task.kernel:sha1 for client.0 is None
```
`client.0` is `ceph-fuse` and i don't think have package of it for jammy

Comment (dparmar18) [reply qa/suites/fs/mixed-clients/distro/ubuntu/latest.yaml:1]: Yeah

**Comment by kotreshhr** [comment]
@dparmar18 Could please you share the test results with the recent changes ?

**Comment by vshankar** [comment]
@dparmar18 I recall you mentioned in IRC that you ran into failures again after switching to 22.04. What was that?

**Comment by vshankar** [comment]
> @vshankar @kotreshhr http://pulpito.front.sepia.ceph.com/dparmar-2022-09-16_11:10:03-fs:mixed-clients-main-distro-default-smithi/, after latest changes and including 22.04 yaml, it still fails with the same error

I will have a look.

**Comment by vshankar** [comment]
> > @vshankar @kotreshhr http://pulpito.front.sepia.ceph.com/dparmar-2022-09-16_11:10:03-fs:mixed-clients-main-distro-default-smithi/, after latest changes and including 22.04 yaml, it still fails with the same error
> 
> I will have a look.

So no ceph-fuse pkgs for 22.04 as per http://pulpito.front.sepia.ceph.com/dparmar-2022-09-16_11:10:03-fs:mixed-clients-main-distro-default-smithi/7034984

IDK. @batrick?

**Comment by batrick** [comment]
Why not just link to `qa.cephfs/distros`? I'm not sure why we need to use 22.04?

**Comment by vshankar** [comment]
> Why not just link to `qa.cephfs/distros`? I'm not sure why we need to use 22.04?

Shaman's builds pkgs for 22.04. Maybe I'm missing the point of why not to use 22.04.

**Comment by dparmar18** [comment]
So basically it seems like API URL issue, please try running this code:

```
import requests


def query_url(api_url, limit_break=60):
    i = 0
    while True:
        response = requests.get(api_url)
        if len(response.json()) == 0:
            print(f"Empty response after {i} tries")
            break
        if i == limit_break:
            print(f"{limit_break} non-stop queries successful! Stopping...")
            break
        i += 1
        print(f"{i} queries successful")


query_url("https://shaman.ceph.com/api/search?status=ready&project=ceph&flavor=default&distros=ubuntu%2F22.04%2Fx86_64&sha1=4ba8783ccf9523c844609cf31d11222e09f04c26",
          limit_break=100)
```

With ubuntu jammy URL, it would failing in few tries(< 10 mostly) while for ubuntu focal it would hit `limit_break`.

```
[dparmar@dparmar-fedora demo]$ python queryapi.py 
1 queries successful
2 queries successful
Empty response after 2 tries
[dparmar@dparmar-fedora demo]$ python queryapi.py 
1 queries successful
Empty response after 1 tries
[dparmar@dparmar-fedora demo]$ python queryapi.py 
1 queries successful
2 queries successful
3 queries successful
Empty response after 3 tries
```

---
I tried this manually, when I opened the URL for first time:
Response

When refreshed the tab:
No-Response

**Comment by vshankar** [comment]
> With ubuntu jammy URL, it would failing in few tries(< 10 mostly) while for ubuntu focal it would hit `limit_break`.

Ugh! Intermittent empty response? For 20.04, you don't see such fetch failures?

**Comment by dparmar18** [comment]
> > With ubuntu jammy URL, it would failing in few tries(< 10 mostly) while for ubuntu focal it would hit `limit_break`.
> 
> Ugh! Intermittent empty response? For 20.04, you don't see such fetch failures?

Nope, tested 20.04 URL multiple times and it would successfully query whatever times I want to.

**Comment by vshankar** [comment]
> > > With ubuntu jammy URL, it would failing in few tries(< 10 mostly) while for ubuntu focal it would hit `limit_break`.
> > 
> > 
> > Ugh! Intermittent empty response? For 20.04, you don't see such fetch failures?
> 
> Nope, tested 20.04 URL multiple times and it would successfully query whatever times I want to.

That's weird. Could you being this up in #sepia or in sepia ML?

**Comment by dparmar18** [comment]
> > > > With ubuntu jammy URL, it would failing in few tries(< 10 mostly) while for ubuntu focal it would hit `limit_break`.
> > > 
> > > 
> > > Ugh! Intermittent empty response? For 20.04, you don't see such fetch failures?
> > 
> > 
> > Nope, tested 20.04 URL multiple times and it would successfully query whatever times I want to.
> 
> That's weird. Could you being this up in #sepia or in sepia ML?

Yeah, sure

**Comment by dmick** [comment]
I reproduced the problem; it's very spotty.  Sometimes 1, sometimes 4 before an "empty" response, with error code 200, so, no error.  Twice, without changing anything, I ran to 100 iterations.  So far I don't see anything unusual in the logs.

**Comment by dparmar18** [comment]
> I reproduced the problem; it's very spotty. Sometimes 1, sometimes 4 before an "empty" response, with error code 200, so, no error. Twice, without changing anything, I ran to 100 iterations. So far I don't see anything unusual in the logs.

@dmick It undoubtedly sends the response every time but when JSON is empty teuthology can't find something like `"version": "18.0.0-144-g4ba8783c"` that it needs and fails at `raise VersionNotFoundError(self._result.url)`. Check [this](http://qa-proxy.ceph.com/teuthology/dparmar-2022-09-29_18:06:26-fs:mixed-clients-main-distro-default-smithi/7048359/teuthology.log) log out.

**Comment by dmick** [comment]
No, I understand the error, and it's definitely intermittent.  What I meant was I couldn't find anything abnormal in the shaman logs (yet).

**Comment by ChristinaMeno** [comment]
@bigjust @gabriellasroman @rakshithakamath94 FYI There's a intermittent bug around shaman being discussed here

**Comment by dmick** [comment]
I restarted 2.shaman with a debug log message in place to narrow down the issue....and now I can't reproduce it.  :(

I'll file another bug to track the shaman issue, but it should not affect this (or other) PRs at the moment.

**Comment by dparmar18** [comment]
@vshankar after carrying out a few teuthology runs for testing the PR: (1) RHEL jobs would fail and (2) https://tracker.ceph.com/issues/57655 is experienced with ubuntu 22.04 jobs.

(1) is understandable to due to the current RHEL Satellite issue, but (2) is something interesting. I tried to dig into the issue, I'm not sure if the kernels are being compiled with AS=clang or not but I found relevant patch already in the code https://github.com/ceph/ceph-client/commit/e81e0724432542af8d8c702c31e9d82f57b1ff31 which can also be found in https://github.com/torvalds/linux/commit/e81e0724432542af8d8c702c31e9d82f57b1ff31. So if this is still happening, there is something else that needs to be looked at. Any idea what could be the issue?

**Comment by vshankar** [comment]
> @vshankar after carrying out a few teuthology runs for testing the PR: (1) RHEL jobs would fail and (2) https://tracker.ceph.com/issues/57655 is experienced with ubuntu 22.04 jobs.
> 
> (1) is understandable to due to the current RHEL Satellite issue, but (2) is something interesting. I tried to dig into the issue, I'm not sure if the kernels are being compiled with AS=clang or not but I found relevant patch already in the code [ceph/ceph-client@e81e072](https://github.com/ceph/ceph-client/commit/e81e0724432542af8d8c702c31e9d82f57b1ff31) which can also be found in [torvalds/linux@e81e072](https://github.com/torvalds/linux/commit/e81e0724432542af8d8c702c31e9d82f57b1ff31). So if this is still happening, there is something else that needs to be looked at. Any idea what could be the issue?
> 
> EDIT: Maybe we can skip testing this PR till this issue is sorted?

I would like to merge this change so that the actual MDS bug shows up in tests and is worked on :)

**Comment by dparmar18** [comment]
> > @vshankar after carrying out a few teuthology runs for testing the PR: (1) RHEL jobs would fail and (2) https://tracker.ceph.com/issues/57655 is experienced with ubuntu 22.04 jobs.
> > (1) is understandable to due to the current RHEL Satellite issue, but (2) is something interesting. I tried to dig into the issue, I'm not sure if the kernels are being compiled with AS=clang or not but I found relevant patch already in the code [ceph/ceph-client@e81e072](https://github.com/ceph/ceph-client/commit/e81e0724432542af8d8c702c31e9d82f57b1ff31) which can also be found in [torvalds/linux@e81e072](https://github.com/torvalds/linux/commit/e81e0724432542af8d8c702c31e9d82f57b1ff31). So if this is still happening, there is something else that needs to be looked at. Any idea what could be the issue?
> > EDIT: Maybe we can skip testing this PR till this issue is sorted?
> 
> I would like to merge this change so that the actual MDS bug shows up in tests and is worked on :)

Yeah! +1

**Review by rishabh-d-dave — APPROVED** [review]
https://tracker.ceph.com/projects/cephfs/wiki/Main#30-JAN-2023

**Review by vshankar — APPROVED** [review]

**Pull request merged**